### PR TITLE
chore: separation of UT and DT releases

### PR DESCRIPTION
**Pull request by sanpj2292**
## Description of the change

Planning to separate the releases of user-transformation and destination transformation. Because there is no hard dependency between both, independence in releases provides greater control and reduction build times for destination transformation

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by koladilip — APPROVED** [review]

File: .github/workflows/prepare-for-prod-deploy.yml [hunk 1]
```diff
@@ -138,29 +116,9 @@ jobs:
 
           cd ../../config-be-rudder-transformer
           yq eval -i ".config-be-rudder-transformer.image.tag=\"$TAG_NAME\"" values.prod.yaml
-          yq eval -i ".config-be-user-transformer.image.tag=\"$TAG_NAME\"" values.prod.yaml
-          yq eval -i ".config-be-user-transformer.image.repository=\"$TF_IMAGE_REPOSITORY\"" values.prod.yaml
```
Comment (krishna2020) [inline .github/workflows/prepare-for-prod-deploy.yml:142]: TF_IMAGE_REPOSITORY should stay

File: .github/workflows/prepare-for-prod-deploy.yml [hunk 2]
```diff
@@ -138,29 +116,9 @@ jobs:
...
           git add values.prod.yaml
 
           git commit -m "chore: upgrade shared transformers to $TAG_NAME"
           git push -u origin shared-transformer-$TAG_NAME
 
           hub pull-request -m "chore: upgrade shared transformers to $TAG_NAME"
-
-      - name: Update Helm Chart and Raise Pull Request For Hosted Transformer
```
Comment (krishna2020) [inline .github/workflows/prepare-for-prod-deploy.yml:150]: this action should stay for updating transformer image in hosted deployment

File: .github/workflows/prepare-for-prod-deploy.yml [hunk 3]
```diff
@@ -138,29 +116,9 @@ jobs:
...
-        env:
-          GITHUB_TOKEN: ${{ secrets.PAT }}
-        run: |
-          cd rudder-devops
-          git checkout -b hosted-transformer-$TAG_NAME ${{steps.extract_branch_name.outputs.branch_name}}
-
-          cd customer-objects/multi-tenant-us
-          yq eval -i ".spec.transformer.image.version=\"$TAG_NAME\"" hostedmtedmt.yaml
-          yq eval -i ".spec.user_transformer.image.version=\"$UT_TAG_NAME\"" hostedmtedmt.yaml
```
Comment (krishna2020) [inline .github/workflows/prepare-for-prod-deploy.yml:159]: Removal of this line is sufficient

File: .github/workflows/prepare-for-prod-ut-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+      - name: Update Helm Chart and Raise Pull Request For Hosted Transformer
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}
+        run: |
+          cd rudder-devops
+          git checkout -b hosted-user-transformer-$TAG_NAME ${{steps.extract_branch_name.outputs.branch_name}}
+
+          cd customer-objects/multi-tenant-us
+          yq eval -i ".spec.transformer.image.version=\"$TAG_NAME\"" hostedmtedmt.yaml
```
Comment (krishna2020) [inline .github/workflows/prepare-for-prod-ut-deploy.yml:127]: Remove this once implemented following
https://github.com/rudderlabs/rudder-transformer/pull/2332#discussion_r1255617793

File: .github/workflows/prepare-for-prod-deploy.yml (hunk 1 above)
Comment (krishna2020) [reply .github/workflows/prepare-for-prod-deploy.yml:142]: but with config-be-rudder-transformer.image.repository

**Comment by krishna2020** [comment]
@sanpj2292 Applied your commit on latest cut from develop branch & addressed comments - https://github.com/rudderlabs/rudder-transformer/pull/2350

**Pull request closed without merging**